is_core/generic_views/form_views.py

```diff
@@ -308,54 +308,55 @@ def generate_readonly_fields(self):
     def get_inline_views(self):
         return self.inline_views or ()
 
+    def generate_inline_views(self):
+        inline_views = self.get_inline_views()
+        fieldsets = self.get_fieldsets()
+        if inline_views and fieldsets:
+            raise ImproperlyConfigured('You can define only one of inline views and fieldsets')
+        return inline_views or get_inline_views_from_fieldsets(fieldsets)
+
     def init_inline_views(self, instance):
-        inline_views = OrderedDict()
-        for inline_view in self.get_inline_views():
-            inline_views[inline_view.__name__] = inline_view(self.request, self, instance)
-        return inline_views
+        return [inline_view(self.request, self, instance) if isinstance(inline_view, type) else inline_view
+                for inline_view in self.generate_inline_views()]
 
     def _filter_inline_form_views(self, inline_views):
-        inline_form_views = OrderedDict()
-        for name, view in inline_views.items():
-            if isinstance(view, InlineFormView):
-                inline_form_views[name] = view
-        return inline_form_views
+        return [view for view in inline_views if isinstance(view, InlineFormView)]
 
-    def generate_fieldsets(self):
-        fieldsets = self.get_fieldsets()
+    def generate_fieldsets(self, inline_views=None, **kwargs):
+        fieldsets = deepcopy(self.get_fieldsets())
```

Comment:
schválně dělám deepcopy jelikož dict fieldsetů budu upravovat. Potřebuji nějakým způsobem předat inicializované inline views do fieldsetu tak abych je mohl vykreslit ve formuláři (už se to nepředává ve dvou datových objektech, které byly provázané klíčem (názvem inlinu) to nedávalo smysl)